subsys/bluetooth/host/id.c

```diff
@@ -36,6 +36,8 @@ struct bt_adv_id_check_data {
 #if defined(CONFIG_BT_OBSERVER) || defined(CONFIG_BT_BROADCASTER)
 const bt_addr_le_t *bt_lookup_id_addr(uint8_t id, const bt_addr_le_t *addr)
 {
+	__ASSERT_NO_MSG(addr != NULL);
```

Comment:
I would suggest to use `CHECKIF` instead of `__ASSERT_NO_MSG` when verifying API input values